ign_rviz_plugins/src/rviz/plugins/CovarianceVisual.cpp

```diff
@@ -0,0 +1,519 @@
+// Copyright (c) 2022 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ignition/rviz/plugins/CovarianceVisual.hpp"
+
+// diagonalization utilities
+namespace
+{
+double deg2rad(double degrees)
+{
+  return degrees * 3.14159265358979 / 180.0;
+}
+
+  // Local function to force the axis to be right handed for 3D. Taken from ecl_statistics
+void makeRightHanded(Eigen::Matrix3d & eigenvectors, Eigen::Vector3d & eigenvalues)
```

Comment:
can you do this math with `ign-math` ? I would perfer not to add Eigen as a dependency

If there is something not in the API we might consider to add it there